src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-Vector.java.template

```diff
@@ -457,6 +457,85 @@ public abstract class $abstractvectortype$ extends AbstractVector<$Boxtype$> {
         return ($type$)((((($type$)a) & $Boxtype$.toUnsignedInt(($type$)-1)) >>> (n & $Boxtype$.SIZE-1)) | (((($type$)a) & $Boxtype$.toUnsignedInt(($type$)-1)) << ($Boxtype$.SIZE - (n & $Boxtype$.SIZE-1))));
 #end[intOrLong]
     }
+
+#if[intOrLong]
+    /* Implementation note: The implementation is based on Compress or Generalized Extract mentioned in
+     * Henry S. Warren, Jr's Hackers Delight, Addison Wesley, 2002.
+     */
+    static $type$ compressBits($type$ a, $type$ mask) {
```

Comment:
No need to make these part of the template. Copy from https://github.com/openjdk/jdk/pull/8115 as Sandhya indicated and place them in a package private separate class, such as `CompressExpand`. Do the same for the tests and use directly. Then the code will be easier to replace.